### ISSUE TITLE
Issue connect with Cyberduck

### ISSUE THREAD

**Issue by sham1111** [kolban-google/sftp-gcs#5]
Hi,
thanks for the great tutorial. I have setup virtual machine exactly the same according to you instructions .I managed to connect google storage from virtual machine but when I tried to connect from cyberduck its not connected. This is the error . **Operation timed out. The connection attempt was rejected. The server may be down, or your network may not be properly configured**   In the tutorial you have not explained any firewall related issue:-)
thanks in advance

**Comment by kolban-google** [comment]
Howdy my friend ... let's see if we can get a handle on the puzzle.  I'm assuming that you are running the sftp demon/code on a Compute Engine within GCP.  Where are you running your cyberduck client?  Is the puzzle only with the cyberduck client?  If you tried with an alternate SFTP client does it work?

**Comment by sham1111** [comment]
thanks for the quick reply, Yes I am running demon/ code on a computer engine and cyber duck on the local machine. No it does not work with alternative SFTP client as well. I think port is the issue and I tried to create firewall rule on GCP but it still does not work. I like the way you did the implementation and it's perfect:-) When I run locally in Compute engine it works.If we can solve this access outside with username and password.This is great implementation

**Comment by kolban-google** [comment]
Okly ... I'm imagining you are running the demon on a Compute Engine on GCP.  I am imagining that the Compute Engine is running a Linux OS such as Debian.  I am imagining that the Compute Engine is attached to a GCP VPC network subdomain.  I am also imagining that the Compute Engine has a public IP address that you are using to reach it over the Internet.

Have you created any firewall rules for accessing the demon?  What TCP/IP port number is the demon listening upon?

My gut is saying that we will have to create an inbound (ingress) firewall rule that permits traffic to come into your Compute Engine from the Internet to the port on which the demon is listening.

**Comment by sham1111** [comment]
yes, I follow the same instruction according to your tutorial.Yes,compute engine is attached and using external IP to access from cyber duck. 
I have created the rule for firewall rules to the demon and demon listening port is 8022
Everything as you recommended:-) Here are the commonds:
sudo apt-get update
sudo apt-get install nodejs git --yes
curl -L http://www.npmjs.org/install.sh | sudo sh
npm install
git clone https://github.com/kolban-google/sftp-gcs.git
cd sftp-gcs
ls
npm install
node sftp-gcs --debug=debug --bucket gs://gubi_oddo --port=8022 --user=***** --password=*********
If you have a time and I can share my screen on skype or somewhere else?

**Comment by sham1111** [comment]
Image with firewall rule
![image](https://user-images.githubusercontent.com/81377476/112532909-ad373d00-8da9-11eb-84b9-162ed69a61ef.png)

**Comment by kolban-google** [comment]
I think a screen share will be great.  Its 3:18pm on Thursday afternoon.  You can email me at kolban@google.com or try and use Google Chat at the same address.  I have a client call 4pm-5pm.  I am out tomorrow ... if not now, then maybe we can synch up for next week?  Email me a couple of good times for you as needed.

Neil

**Comment by sham1111** [comment]
@kolban-google  thanks and it works after opening the firewall:-) Thanks for the help

**Comment by GillesFerone** [comment]
Hi there,
Thank you for the tutorial and elegant bridging solution you built @kolban-google.
I'm encountering a similar issue.
I followed all the steps laid out here https://www.youtube.com/watch?v=yREQRw1H23Q&t=389s
Trying to connect to the SFTP server from my mac computer via Cyberduck fails with the same error message you laid out @sham1111 
**Operation timed out. The connection attempt was rejected. The server may be down, or your network may not be properly configured**
I didn't set any specific firewall rule.
Could you expand on this **it works after opening the firewall** ?
I tried using the native "SFTP" of my mac terminal and was unable to connect to the server as well..

**Comment by kmeljko** [comment]
After Filezilla I found that cyberduck also can't be used for sftp with this software (linux ftp command and winscp works). Is the problem same as with Filezilla with serial or something different
